app/models/repo.rb

```diff
@@ -3,14 +3,24 @@ class Repo
   base_uri "docs.publishing.service.gov.uk"
 
   def self.all
-    response ||= get("/apps.json")
-    JSON.parse(response.body)
-  rescue HTTParty::Error, JSON::ParserError => e
-    Rails.logger.debug "Error fetching govuk repos: #{e.message}"
-    []
+    Rails.cache.fetch("apps_json_data", expires_in: 12.hours) do
+      response ||= get("/apps.json")
+      JSON.parse(response.body)
+    rescue HTTParty::Error, JSON::ParserError => e
+      Rails.logger.debug "Error fetching govuk repos: #{e.message}"
+      []
+    end
   end
 
   def self.find_by(app_name:)
     all.find { |app| app["app_name"] == app_name.parameterize }
   end
+
+  def self.get_url(app_name:)
```

Comment:
I wonder if those methods should be called just `url`, `shortname` to follow the convention of other models. For example we have `Application.status` not `Application.get_status`.

```suggestion
  def self.url(app_name:)
```

Reply:
Ooh yes good shout thanks @AgaDufrat !